src/network/event_loop.rs

```diff
@@ -365,6 +370,55 @@ impl EventLoop {
 						"AutoNat Old status: {:#?}. AutoNat New status: {:#?}",
 						old, new
 					);
+					// check if went private or are private
+					// if so, create reservation request with relay
+					if new == NatStatus::Private {
+						info!("Autonat says we're still private.");
+						// choose relay by random
+						if let Some(relay) = self.relay_nodes.choose(&mut rand::thread_rng()) {
+							let (relay_id, relay_addr) = relay.clone();
+							// appoint this relay as our chosen one
+							self.relay_reservation(relay_id, relay_addr);
+						}
+					};
+				},
+			},
+			SwarmEvent::Behaviour(BehaviourEvent::Relay(event)) => match event {
```

Comment:
Are these all relay events (to be extracted to a separate relay repo at some point in the near future) or client ones?

Reply:
separations will happen
anything that isn't a relay should have its client events, and relays should only be concerned with their own events